Widgets/Widgets/TopReadWidget.swift

```diff
@@ -54,14 +54,19 @@ final class TopReadData {
 
         // The WMFContentGroup can only be accessed synchronously
         // re-accessing it from the main queue or another queue might lead to unexpected behavior
+        let languageVariant = topRead.variant
         let layoutDirection: LayoutDirection = topRead.isRTL ? .rightToLeft : .leftToRight
         let groupURL = topRead.url
         let isCurrent = topRead.isForToday // even though the top read data is from yesterday, the content group is for today
         var rankedElements: [TopReadEntry.RankedElement] = []
+
         for article in results {
+            article.articleURL.wmf_languageVariantCode = languageVariant
             if let articlePreview = dataStore.fetchArticle(with: article.articleURL) {
                 if let viewCounts = articlePreview.pageViewsSortedByDate {
-                    rankedElements.append(.init(title: article.displayTitle, description: article.wikidataDescription ?? article.snippet ?? "", articleURL: article.articleURL, thumbnailURL: article.thumbnailURL, viewCounts: viewCounts))
+                    let title = articlePreview.displayTitle ?? article.displayTitle
```

Comment:
I was initially confused because the PR notes say that the title and description of the article would be preferred over those values in the article preview, but the code seems to indicate the opposite. I see that the logic is correct, but the variable names seem to indicate the opposite.

Maybe consider swapping the names 'article' and 'articlePreview' so that 'articleResult' refers to the `WMFFeedTopReadArticlePreview` and 'article' refers to the 'WMFArticle'.

Reply:
The naming here is indeed a mess. Might as well use this as an opportunity to clean it up. Updated.